applications/asset_tracker/src/main.c

```diff
@@ -1354,22 +1573,38 @@ void main(void)
 		}
 
 		if ((fds[0].revents & POLLNVAL) == POLLNVAL) {
-			if (atomic_get(&reconnect_to_cloud)) {
-				k_delayed_work_cancel(&cloud_reboot_work);
-				LOG_INF("Attempting reconnect...");
+			if (atomic_get(&cloud_association) ==
+					CLOUD_ASSOCIATION_STATE_RECONNECT) {
+				reconnecting = true;
```

Comment:
Will this currently work? I haven't tested lately, but it didn't before NCS 1.2.0 without a hack in nrf_cloud_transport due to issue in upstream MQTT. It could be okay now after MQTT disconnect procedure was fixed a bit before the release

Reply:
yes it has been working for me...  tested it quite a bit 😅 
the only thing i couldn't get to work was a reconnect after a pollhup when using the lwm2m carrier lib... so the restart still occurs in that case.